src/phylum/ci/ci_base.py

```diff
@@ -487,6 +495,77 @@ def _ensure_project_exists(self) -> None:
         if self._project_file_already_existed:
             LOG.warning("Overwrote previous `.phylum_project` file found at: %s", self._phylum_project_file)
 
+    def _set_repo_url(self) -> None:  # noqa: C901 ; can remove after https://github.com/phylum-dev/cli/issues/1300
+        """Set the repository URL for the project.
+
+        The value is meant to be the Web UI URL for where the project is hosted.
+        It should not be set or changed if a value already exists.
+        """
+        if self.repo_url is None:
+            LOG.debug("Repository URL not available to set")
+            return
+
+        # This block can be simplified after https://github.com/phylum-dev/cli/issues/1300
+        LOG.info("Checking project for existing repository URL ...")
+        LOG.debug("Finding ID for Phylum project: %s", self.phylum_project)
+        cmd = [str(self.cli_path), "project", "list", "--json"]
+        if self.phylum_group:
+            LOG.debug("Using Phylum group: %s", self.phylum_group)
+            cmd.extend(["--group", self.phylum_group])
+        try:
+            projects_list = subprocess.run(cmd, check=True, capture_output=True, text=True).stdout.strip()  # noqa: S603
+        except subprocess.CalledProcessError as err:
+            pprint_subprocess_error(err)
+            msg = """\
+                Phylum project listing failed. Skipping repository URL check. Use CLI
+                to manually set it: https://docs.phylum.io/docs/phylum_project_update"""
+            LOG.warning(textwrap.dedent(msg))
+            return
+        projects: list = json.loads(projects_list)
+        project_id, repo_url = None, None
+        project: dict
+        for project in projects:
+            if project.get("name") == self.phylum_project:
+                project_id = project.get("id")
+                repo_url = project.get("repository_url")
+                break
```

Comment:
I'm no Pythonista, but can this work?
```python
project = next(filter(lambda p: p.get("name") == self.phylum_project, projects), None)
```

I tested it out on the playground and it seems to have the same early return behavior.

Reply:
Suggestion accepted...with minor changes to make the type hint checkers happy.